tastypie/resources.py

```diff
@@ -1430,6 +1430,9 @@ def put_detail(self, request, **kwargs):
             if not self._meta.always_return_data:
                 return http.HttpNoContent()
             else:
+                # Invalidate prefetched_objects_cache for bundled object
+                # because we might have changed a prefetched field
+                updated_bundle.obj._prefetched_objects_cache = {}
```

Comment:
Sorry for the delay ... when adding prefetch_related to the queryset (resources.py:213) to reduce the number of db requests, tastypie will return the prefetched results for M2M fields, even though we updated them in the request.